src/Heading/Heading.js

```diff
@@ -1,77 +0,0 @@
-import styled from '@emotion/styled';
-import { css } from '@emotion/react';
-
-import Text from '../Text';
-
-const lineHeightMultiplier = 0.5;
-
-const baseStyles = ({ theme }) => css`
-  line-height: ${theme.lineHeights.small};
-`;
-
-const calculatedMargin = ({ mb, fontSize, theme }) =>
-  !mb &&
-  css`
-    margin-bottom: calc(${theme.fontSizes[fontSize]} * ${lineHeightMultiplier});
-  `;
-
-const Heading = styled(Text)`
-  ${baseStyles};
-  ${calculatedMargin};
-`;
-
-Heading.defaultProps = {
-  as: 'h1',
-  color: 'text.dark',
-  fontWeight: 'bold',
-  lineheight: 'small'
```

Comment:
Note the typo here: `lineheight` should be `lineHeight`.  This is fixed in the TS version below (caught thanks to TypeScript as there's no `lineheight` prop), and is responsible for several of the snapshot diffs.